main.tf

```diff
@@ -307,6 +269,67 @@ resource "ibm_cos_bucket" "cos_bucket1" {
   }
 }
 
+locals {
+  expiration_or_archiving_rule_enabled = (length(local.expire_enabled) != 0 || length(local.archive_enabled) != 0)
+
+  create_cos_bucket  = (var.kms_encryption_enabled && var.create_cos_bucket) ? true : false
+  create_cos_bucket1 = (!var.kms_encryption_enabled && var.create_cos_bucket) ? true : false
+
+  cos_bucket_resource = local.create_cos_bucket ? ibm_cos_bucket.cos_bucket : local.create_cos_bucket1 ? ibm_cos_bucket.cos_bucket1 : null
+
+  ## Only one of these values can be set, leaving 2 of 3 null, compact function removes nulls.
+  ## We then take the only value left in the list
+  cos_region = compact([var.region, var.cross_region_location, var.single_site_location])[0]
+}
+
+resource "time_sleep" "wait_for_cos_bucket_lifecycle" {
+  count           = (local.create_cos_bucket || local.create_cos_bucket1) && local.expiration_or_archiving_rule_enabled ? 1 : 0
+  create_duration = "30s"
+}
+
+resource "ibm_cos_bucket_lifecycle_configuration" "cos_bucket_lifecycle" {
+  count = (local.create_cos_bucket || local.create_cos_bucket1) && local.expiration_or_archiving_rule_enabled ? 1 : 0
+
+  depends_on = [time_sleep.wait_for_cos_bucket_lifecycle]
+
+  bucket_crn      = local.cos_bucket_resource[count.index].crn
+  bucket_location = local.cos_region
+
+  dynamic "lifecycle_rule" {
+    ## This for_each block is NOT a loop to attach to multiple expiration blocks.
+    ## This block is only used to conditionally add expiration block depending on expire rule is enabled.
+    for_each = local.expire_enabled
+    content {
+      expiration {
+        days = var.expire_days
+      }
+      filter {
+        prefix = ""
+      }
+      rule_id = "expiry-rule"
+      status  = "enable"
+    }
+  }
+  dynamic "lifecycle_rule" {
+    ## This for_each block is NOT a loop to attach to multiple transition blocks.
+    ## This block is only used to conditionally add retention block depending on archive rule is enabled.
+    for_each = local.archive_enabled
+    content {
+      transition {
+        days = var.archive_days
+        ## The new values changed from Capatalized to all Upper case, avoid having to change values in new release
+        storage_class = upper(var.archive_type)
+
+      }
+      filter {
+        prefix = ""
```

Comment:
we probably should expose this to allow users to set it? perhaps add an optional `archive_filter_prefix` input? Default it to `null` (and have our code add empty string if thats the case)